playbooks/common/openshift-cluster/update_repos_and_packages.yml

```diff
@@ -5,8 +5,10 @@
   roles:
   - role: rhel_subscribe
     when: deployment_type in ["enterprise", "atomic-enterprise", "openshift-enterprise"] and
-          ansible_distribution == "RedHat" and
-          lookup('oo_option', 'rhel_skip_subscription') | default(rhsub_skip, True) |
-            default('no', True) | lower in ['no', 'false']
+          ansible_distribution == "RedHat" and not ( openshift_rhel_skip_subscription |
+          default(lookup('oo_option', 'rhel_skip_subscription')) | default(rhsub_skip) |
+          default(False) | bool )
```

Comment:
might make sense to break some of this out into the vars section to clean up this when clause a bit.

Also, the value of rhsub_skip and the default(False) will never be used here.

Warning, the below is completely untested:
```
vars:
...
  # store this value to conditionally set it only if it doesn't return '', since the oo_option
  # lookup filter will return '' if the variable isn't set.
  t_oo_opt_skip_sub: "{{ lookup('oo_option', 'rhel_skip_subscription') }}"
  # to maintain the ordering you indicated above, we need to work around the fact that
  # the oo_option lookup plugin will return '' if the env var/variable is not defined.
  t_skip_sub_val: "{{ openshift_rhel_skip_subscription
                                | default(t_oo_opt_skip_sub if t_oo_opt_skip_sub != '' else rhsub_skip | default(False)) }}"
  t_valid_dt_for_sub: "{{ deployment_type in ['enterprise', 'atomic-enterprise', 'openshift-enterprise'] }}"
  t_skip_rhel_subscribe: "{{ t_valid_dt_for_sub | bool and ansible_distribution == 'RedHat' and not t_skip_sub_val | bool }}"
```

It can be cleaned up a bit further, by using a conditional set_fact task to define openshift_rhel_skip_subscription as the value of the oo_option lookup on rhel_subscription if it isn't already defined.
```
vars:
...
  t_oo_opt_rhel_skip_sub: "{{ lookup('oo_option', 'rhel_skip_subscription') }}"
  t_valid_dt_for_sub: "{{ deployment_type in ['enterprise', 'atomic-enterprise', 'openshift-enterprise'] }}"
...
pre_tasks:
- set_fact:
    openshift_rhel_skip_subscription: "{{ t_oo_opt_rhel_skip_sub if t_oo_opt_rhel_skip_sub != '' else rhsub_skip | default(False) }}"
  when: openshift_rhel_skip_subscription is not defined
...
roles:
- role: rhel_subscribe:
  when: t_valid_dt_for_sub | bool and ansible_distribution == 'RedHat' and not openshift_rhel_skip_subscription | bool
```